variables.database.tf

```diff
@@ -0,0 +1,121 @@
+      email_account_admins       = optional(string, "Disabled")
+      email_addresses            = optional(list(string))
+      retention_days             = optional(number)
+      storage_account_access_key = optional(string)
+      storage_endpoint           = optional(string)
+    }))
+
+
+    tags = optional(map(string))
+  }))
+
+  validation {
+    condition = can([
+      for database, config in var.databases : (
+        config.create_mode == null ? true : contains([
+          "Copy", "Default", "OnlineSecondary", "PointInTimeRestore", "Recovery", "Restore", "RestoreExternalBackup",
+          "RestoreExternalBackupSecondary", "RestoreLongTermRetentionBackup", "Secondary"
+        ], config.create_mode)
+      )
+    ])
+    error_message = "Invalid value for create_mode. Allowed values are Copy, Default, OnlineSecondary, PointInTimeRestore, Recovery, Restore, RestoreExternalBackup, RestoreExternalBackupSecondary, RestoreLongTermRetentionBackup, or Secondary."
+  }
+
+  validation {
+    condition = can([
+      for database, config in var.databases : (
+        config.maintenance_configuration_name == null ? true : contains([
+          "SQL_Default",
+          "SQL_AustraliaEast_DB_1", "SQL_AustraliaEast_DB_2", "SQL_AustraliaSoutheast_DB_1", "SQL_AustraliaSoutheast_DB_2",
+          "SQL_BrazilSoutheast_DB_1", "SQL_BrazilSoutheast_DB_2", "SQL_BrazilSouth_DB_1", "SQL_BrazilSouth_DB_2",
+          "SQL_CanadaCentral_DB_1", "SQL_CanadaCentral_DB_2", "SQL_CanadaEast_DB_1", "SQL_CanadaEast_DB_2",
+          "SQL_CentralIndia_DB_1", "SQL_CentralIndia_DB_2", "SQL_CentralUS_DB_1", "SQL_CentralUS_DB_2",
+          "SQL_EastAsia_DB_1", "SQL_EastAsia_DB_2", "SQL_EastUS_DB_1", "SQL_EastUS_DB_2", "SQL_EastUS2_DB_1", "SQL_EastUS2_DB_2",
+          "SQL_FranceCentral_DB_1", "SQL_FranceCentral_DB_2", "SQL_FranceSouth_DB_1", "SQL_FranceSouth_DB_2",
+          "SQL_GermanyWestCentral_DB_1", "SQL_GermanyWestCentral_DB_2",
+          "SQL_JapanEast_DB_1", "SQL_JapanEast_DB_2", "SQL_JapanWest_DB_1", "SQL_JapanWest_DB_2",
+          "SQL_NorthCentralUS_DB_1", "SQL_NorthCentralUS_DB_2", "SQL_NorthEurope_DB_1", "SQL_NorthEurope_DB_2",
+          "SQL_SoutheastAsia_DB_1", "SQL_SoutheastAsia_DB_2", "SQL_SouthCentralUS_DB_1", "SQL_SouthCentralUS_DB_2", "SQL_SouthIndia_DB_1", "SQL_SouthIndia_DB_2",
+          "SQL_SwitzerlandNorth_DB_1", "SQL_SwitzerlandNorth_DB_2",
+          "SQL_UAENorth_DB_1", "SQL_UAENorth_DB_2", "SQL_UKSouth_DB_1", "SQL_UKSouth_DB_2", "SQL_UKWest_DB_1", "SQL_UKWest_DB_2",
+          "SQL_WestCentralUS_DB_1", "SQL_WestCentralUS_DB_2", "SQL_WestEurope_DB_1", "SQL_WestEurope_DB_2",
+          "SQL_WestUS2_DB_1", "SQL_WestUS2_DB_2", "SQL_WestUS_DB_1", "SQL_WestUS_DB_2"
+        ], config.maintenance_configuration_name)
+      )
+    ])
+    error_message = "Invalid value for maintenance_configuration_name. Allowed values are SQL_Default, SQL_EastUS_DB_1, SQL_EastUS2_DB_1, ..., SQL_UAENorth_DB_2."
```

Comment:
Seems strange to have the `...` here, maybe don't include any.

Reply:
Removed